enhancements/insights/config-api.md

```diff
@@ -0,0 +1,272 @@
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type SimpleContentAccessStatus struct {
+	// LastDownloadTime is the last time when the SCA certs were downloaded.
+	LastDownloadTime time.Time `json:"lastDownloadTime,omitempty"`
+	// Status provides last known status information of the SCA controller.
+	Status string `json:"status,omitempty"`
+}
+
+type ClusterTransferStatus struct {
+	// LastCheckTime is the last time when the ClusterTranfer was checked.
+	LastCheckTime time.Time `json:"lastCheckTime,omitempty"`
+	// Status provides last known status information of the ClusterTransfer controller.
+	Status string `json:"status,omitempty"`
+}
+```
+
+### User Stories
+
+- As an OCP user/cluster-admin I would like to be able to trigger on demand gathering (there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status). 
+- As an OCP user/cluster-admin I would like to have a convenient way to configure Insights.
+- As an OCP user/cluster-admin I would like to see the latest Insights status.
+- As an OCP user/cluster-admin I would like to exclude/include particular gatherers.
+
+### API Extensions
+
+This is adding a new `insights.config.openshift.io` API. The proposal of the structure of the new API is mentioned in the [Proposal](#Proposal) section.
+
+### Risks and Mitigations
+
+Risk: Insights operator is not able to read the new config API.
+
+Mitigation: The default config values are used and they should ensure basic functionality of the Insights operator.
+
+## Design Details
+
+### On demand data gathering options
+
+- schedule it - e.g using something like the `ScheduleGatheringAt` attribute where the user would provide some time in the future.
+  - Pros 
+    - no need to interrupt gathering in progress
+    - we could probably use the `RateLimiting` queue and add a new gathering to the queue with the `AddAfter` method. Otherwise we would need to define some waiting time (when there’s already some gathering in progress) and right now it’s difficult to estimate an average gathering time since the gatherers run in parallel
+  - Cons
+    - user has to define the scheduling time correctly in the future
+    - gathering doesn’t happen immediately
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [roviding a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
```

Comment:
I just learned about another slightly similar option how to trigger an action and that is by setting an annotation. There should be something like that in the CMO so I will try to get more info about it.

Reply:
OK so if I got it correctly the CMO allows users to annotate a PVC. Such PVC is then marked for deletion when rescheduling pods to a different node. 
It would probably mean to annotate the new `Insights` kind (to force gathering) in our case. I think it's better to have the explicit `ForceGatheringReason` attribute in the spec than adding/removing some annotation.